src/platform/context.h

```diff
@@ -21,11 +21,16 @@
 #ifndef AWE_CONTEXT_H
 #define AWE_CONTEXT_H
 
+#include <glm/vec2.hpp>
+
 namespace Platform {
 
 class Context {
 public:
 	virtual void getSize(unsigned int &width, unsigned int &height) = 0;
+	virtual glm::vec2 getSize() = 0;
+	virtual void setSize(unsigned int width, unsigned int height) = 0;
+	virtual void setFullscreen(bool fullscreen) = 0;
```

Comment:
I do not like the these methods in the context interface. It was designed to give the information control the rendering system needs. The rendering system does not need to set the fullscreen and the size. Also there are cases where you do not have this possibility, like for example a game editor (Where the size is determined by the window around it) or a phone screen (where there is at least in Android and iOS no such state like fullscreen).